lib/model.js

```diff
@@ -1252,11 +1252,13 @@ module.exports = (function() {
       // Map field names
       values = Utils.mapValueFieldNames(instance.dataValues, options.fields, this);
 
+      var now = Utils.now(this.sequelize.options.dialect);
+
       if (createdAtAttr && !values[createdAtAttr]) {
-        values[createdAtAttr] = this.__getTimestamp(createdAtAttr);
+        values[createdAtAttr] = this.__getDefaultTimestamp(createdAtAttr) || now;
       }
       if (updatedAtAttr && !values[updatedAtAttr]) {
-        values[updatedAtAttr] = this.__getTimestamp(updatedAtAttr);
+        values[updatedAtAttr] = this.__getDefaultTimestamp(updatedAtAttr) || now;
```

Comment:
That's what getTimestamp does right now. The whole point in my change though is that now will not be in the return value of the function. If you set now twice in a row for updatedAt and createdAt it differs for some parts of a millisecond and produces race condition in our tests.